cpp/src/io/csv/writer_impl.cu

```diff
@@ -472,14 +476,15 @@ void write_csv(data_sink* out_sink,
       //
       std::string delimiter_str{options.get_inter_column_delimiter()};
       auto str_concat_col = [&] {
+        cudf::string_scalar narep{options.get_na_rep(), true, stream};
         if (str_table_view.num_columns() > 1)
-          return cudf::strings::detail::concatenate(str_table_view,
-                                                    delimiter_str,
-                                                    options.get_na_rep(),
-                                                    strings::separator_on_nulls::YES,
-                                                    stream,
-                                                    rmm::mr::get_current_device_resource());
-        cudf::string_scalar narep{options.get_na_rep()};
+          return cudf::strings::detail::concatenate(
+            str_table_view,
+            cudf::string_scalar(delimiter_str, true, stream),
+            narep,
```

Comment:
One parameter is inline defined while the other is not. Please make them consistent.

Reply:
I don't think, it's not a matter of being consistent. That temporary scalar is not required after this usage. So, it's not a named parameter. its scope end after function call.
Here narep is used below this line.